src/send-email.js

```diff
@@ -50,5 +67,135 @@ async function fetchFilesFromRepo() {
     console.error('Error:', error)
   }
 }
+export function filterStale(projects) { // projects is an array of objects
+  const today = new Date();
+  const filtered = [];
+
+  projects.forEach((project) => {
+    const state = project.status;
+    let closeDate = parse(project.opportunityCloses);
+    let startDate = parse(project.startDate);
+    let endDate = parse(project.endDate);
+    let modifyDate = parse(project.lastModified);
+
+    const problems = [];
+
+    switch (state) {
+      default:
+        break;
+      case "open":
+        if (closeDate < today) {
+          problems.push(
+              "The opportunity date has passed, but project is marked open.",
+          );
+        }
+        if (startDate < today) {
+          problems.push("The start date has passed, but project is marked open.");
+        }
+        if (today - modifyDate > 90) {
+          problems.push(
+              "It has been over 90 days since project's last modification.",
+          );
+        }
+        if (endDate < today) {
+          problems.push("The project has ended, but is marked as open.");
+        }
+        break;
+      case "ongoing":
+        if (today - modifyDate > 90) {
+          problems.push(
+              "It has been over 90 days since project's last modification.",
+          );
+        }
+        if (endDate < today) {
+          problems.push("The project has ended, but is marked as ongoing.");
+        }
+        break;
+    }
+
+    if (problems.length > 0) {
+      filtered.push({ ...project, problems: problems });
+    }
+  });
+  return filtered;
+}
+
+export function filterNew(projects, time){ // projects is an array of objects; time is the number of days to search back from day of email
+  const filtered = projects.filter(project => parse(project.created) < subDays(parse(project.created), time));
+  return filtered;
+}
```

Comment:
would it be easier to understand this function if you gave it a date, and just got all the projects created after that date?

Reply:
let's refactor if we end up using it in another spot!